src/main/java/frc/robot/subsystems/Launcher.java

```diff
@@ -0,0 +1,217 @@
+    SmartDashboard.putNumber("Launcher Min Output", LauncherConstants.kMinOutput);
+    SmartDashboard.putNumber("Target Velocity", LauncherConstants.kTargetLaunchVelociy);
+    SmartDashboard.putNumber("Left Speed", m_leftEncoder.getVelocity());
+    SmartDashboard.putNumber("Right Speed", m_rightEncoder.getVelocity());
+
+    //gets values from the dashboard
+    double p = SmartDashboard.getNumber("Launcher P Gain", 0);
+    double i = SmartDashboard.getNumber("Launcher I Gain", 0);
+    double d = SmartDashboard.getNumber("Launcher D Gain", 0);
+    double iz = SmartDashboard.getNumber("Launcher I Zone", 0);
+    double ff = SmartDashboard.getNumber("Launcher Feed Forward", 0);
+    double max = SmartDashboard.getNumber("Launcher Max Output", 0);
+    double min = SmartDashboard.getNumber("Launcher Min Output", 0);
+    double setpoint = SmartDashboard.getNumber("Target Velocity", 0);
+    
+
+    //reset PID constants if changed in dashboard
+    if (p != pVal) {
+      m_pidController.setP(p);
+      pVal = p;
+    }
+
+    if (i != iVal) {
+      m_pidController.setI(i);
+      iVal = i;
+    }
+
+    if (d != dVal) {
+      m_pidController.setD(d);
+      dVal = d;
+    }
+
+    if (iz != izVal) {
+      m_pidController.setIZone(iz);
+      izVal = iz;
+    }
+
+    if (ff != ffVal) {
+      m_pidController.setFF(ff);
+      ffVal = ff;
+    }
+
+    if ((max != maxVal) || (min != minVal)) {
+      m_pidController.setOutputRange(min, max);
+      minVal= min;
+      maxVal = max;  
+    }
+
+    //sets setpoint and control type to PID controller
+    m_pidController.setReference(setpoint, ControlType.kVelocity);
+
+  }
+
+  public void stopPID() {
+    m_pidController.setP(0);
+    m_pidController.setI(0);
+    m_pidController.setD(0);
+    m_pidController.setFF(0);
+    m_pidController.setIZone(0);
+    
+  }
+ 
+  /**
+   * Initialize an encoder.
+   * 
+   * @param encoder The encoder to initialize
+   */
+  private void encoderInit(CANEncoder encoder) {
+    // Set the automatic conversion factor for the encoder. This allows
+    // the encoder to output data in a specified unit.
+    encoder.setVelocityConversionFactor(LauncherConstants.kEncoderSpeedPerPulse);
+    encoderReset(encoder); // Reset the encoder to 0, just in case
+  }
+ 
+  /**
+   * Reset an encoder's position.
+   * 
+   * @param encoder The encoder to reset
+   */
+  private void encoderReset(CANEncoder encoder) {
+    encoder.setPosition(0);
+  }
+ 
+  //sets speed of launcher motors to 0
+  private void stopMotors(CANSparkMax leftLauncher, CANSparkMax rightLauncher) {
+    leftLauncher.set(0);
+    rightLauncher.set(0);
+  }
+ 
+  //get left speed in RPM
+  private double getLeftSpeed() {
+    return m_leftEncoder.getVelocity();
+  }
+ 
+  //get right speed in RPM
+  private double getRightSpeed() {
+    return m_rightEncoder.getVelocity();
+  }
+ 
+  //get average speed
+  private double getSpeed() {
+    return ((getLeftSpeed() + getRightSpeed()) / 2.0);
+  }
```

Comment:
Since we will be using methods like this in most subsystems, the method names should be more descriptive.